lib/galaxy/app.py

```diff
@@ -105,60 +119,68 @@ def __init__(self, **kwargs) -> None:
 
         # Security helper
         self._configure_security()
+        self._register_singleton(IdEncodingHelper, self.security)
+        self._register_singleton(SharedModelMapping, self.model)
+        self._register_singleton(GalaxyModelMapping, self.model)
+        self._register_singleton(scoped_session, self.model.context)
```

Comment:
We can only make this a singleton if we manage to update the scopting function during a request. That would be an elegant solution to the problem though.